### ISSUE TITLE
Not issues but wishes 

### ISSUE THREAD

**Issue by Kabron287** [pavelrevak/rt_soldering_pen#1]
Nice product
1. Accelerometer is highly desirable like in TS100 (SCL/SDA are free)
2. Go to Standby Time adjustable from settings.

**Comment by pavelrevak** [comment]
Adjustable standby timeout is planed feature
Accelerometer , there is no space on PCB, this need redesign.

**Comment by Kabron287** [comment]
No need to redesign. Due to low pin-count it can be easily placed on free PCB space and hand-wired.
You need Just to add the code.

**Comment by pavelrevak** [comment]
yes, it is possible, but not for all devices ,-)

**Comment by Kabron287** [comment]
What all devices? 
On start SW search I2C if found MMA8652FC it use it, else not.

**Comment by pavelrevak** [comment]
Ok, Now I understand what you want.
Yes, this is good idea and personally I also thought about it in the past, but at this moment there is still lot of work with the software. so, I need to postponed the idea for later.

**Comment by pavelrevak** [comment]
But if you need something like "auto-standby" mode, there is still one possibility and without HW change. the current sensor also react on surrounding magnetic field (this magnetic field is compensated and subtracted from final measured current, but this "error" is possible to use for detect near magnetic field, so my idea is to put into stand small magnet which will be placed nearly to the current sensor location and if I detect higher magnetic field I put the iron into standby and if the field will lost I immediately wake up. this will be more easier.

**Comment by Kabron287** [comment]
Good idea for a moment

**Comment by pavelrevak** [comment]
You can see the magnetic field acting on the sensor in menu under: `Current sensor err:`

**Comment by x893** [comment]
Hi,
possible change
1. ACS711ELCTR-12AB-T to ACS711KEXLT-15AB-T
2. LMR14203+1717+PMEG6010CEH to LMZM23600

**Comment by pavelrevak** [comment]
Hi, thanks for suggestions.
Regarding ACS, I previously try to make design with this one, but with QFN it was much harder to layout it on two layer PCB, although it is smaller package, SO8 has so much space under package, so there was not a problem to wire other signals.
And LMZM23600 ... yes, I was looking on this one, but when I started the design it was problem to buy it, because it was very new and also has some special requirements from PCB.

**Comment by Kabron287** [comment]
Made a simple case with the acrylic tube 16mm.
I had to make small recesses for the buttons.
![1](https://user-images.githubusercontent.com/2493191/44219267-49c1e900-a184-11e8-81f6-ecea169a7ac2.jpg)

**Comment by elektrinis** [comment]
+1 for accelerometer. Bit frustrating to find it cold each time.

Also, is it possible to implement supply from USB C PD? There are several examples on github how to setup the USB for 12V 6A and it seems to be fairly simple. This would be extremely useful!
If you could implement USB serial, a bootloader for firmware updates is needed.

**Comment by Kabron287** [comment]
**_There are several examples on github how to setup the USB for 12V 6A_** 
Interesting, would you be kind to give some links?

**Comment by elektrinis** [comment]
Arduino project: https://github.com/ReclaimerLabs/USB_PD
Other: https://github.com/ReclaimerLabs/USB-PD-Breakout
I also saw one for STM, can't find it quickly.

**Comment by Kabron287** [comment]
Something have to be done with the current limitation alhoritm. Due to the very high starting current, most power supplies go into protection. There are not such behaviour in TS100 and TS80.
I guess soft start with variable PWM duty cicle is mandatory.

**Comment by elektrinis** [comment]
I have also noticed that heater is being switched very slowly. You should PWM it at 16kHz, like most power electronics.

**Comment by pavelrevak** [comment]
PWM period is 150ms, there are more reasons for that
1. low EMI
2. I can not measure temperature if power switch is ON, and after OFF I need wait at least 2ms (to stabilise temperature measure amplifier) and then measure for at least 8ms (there is more measurements for averaging) and to keep good duty cycle to maximise power if supply voltage is below 12V I need to make longer PWM.
3. also in every cycle I recalculate all measured values, calculate PID, redraw display with newly measured values, ..

Your power supply MUST withstand 2Ω load (e.g. 6A at 12V or 8A at 16V)
TS100/80 has higher heater resistance, so there is lower current.
I think TS100/80 has also low frequency PWM.

**Comment by elektrinis** [comment]
Thank you for explaining. Not criticizing, just want to understand the issue better.
I have a fair amount of experience with BLDC/AC motor drives that usually run at 16kHz PWM and do much more measurements and calculations than in this case (read on FOC). The trick is to let PWM run on it's own dedicated timer-based peripheral, also use a peripheral to synchronize measurements at start of off state and then have your calculations running independently. Main objective is to free your CPU time by not micromanaging the PWM and only relatively infrequently updating cycle width. When you have high PWM frequency, you don't need to recalculate cycle-by-cycle.
Adding a small capacitor and, perhaps, inductor, will take care of EMI and will also allow to use lower power supply.

**Comment by pavelrevak** [comment]
Yes, thats true, but there is still problem with thermocouple, which give only 8mV at 500°C this has common ground with heating element. and amplifier has some filters, so I need wait some time when it stabilise.
Another problem is that I used MOSFET with very low Rds-on but has one disadvantage that is slower and with higher frequencies will be not fully switched and then it will generate more heat. and this is another problem that I need to keep PCB cool as much as possible, because inside MCU is temperature sensor, which measure second end thermocouple (also reason why MCU is running on 8MHz and not at 48MHz).
I Have plan to try decrease period to about 50ms to have faster response of PID but I will do it later.

Regarding MCU there is also possibility to use STM32L031G6 which is pin compatible and has lower power consumption.

Or if you have some experience and time, you also can try to play with this, source is available.
